### PR TITLE
Made a catalog-grid

### PR DESCRIPTION
**Pull request by SashaViun**
- [DEMO LINK](https://SashaViun.github.io/layout_catalog_grid/)

### CODE REVIEW

File: src/styles/_catalog.scss [hunk 1]
```diff
@@ -1,22 +1,25 @@
...
-  display: flex;
+  display: grid;
+  grid-template: 300px / repeat(4, 200px);
   flex-wrap: wrap;
-
+  gap: 48px;
+  justify-content: center;
+  place-items: center;
   max-width: 944px;
+  padding-bottom: 48px;
```
Comment (OlyaZaiets) [inline src/styles/_catalog.scss:9]: We already have gap, I'm not sure that we need paddings here

File: src/styles/_catalog.scss [hunk 2]
```diff
@@ -1,22 +1,25 @@
 .catalog {
-  display: flex;
+  display: grid;
+  grid-template: 300px / repeat(4, 200px);
   flex-wrap: wrap;
-
+  gap: 48px;
+  justify-content: center;
+  place-items: center;
   max-width: 944px;
```
Comment (OlyaZaiets) [inline src/styles/_catalog.scss:8]: Should we have this style?

File: src/styles/_catalog.scss [hunk 3]
```diff
@@ -1,22 +1,25 @@
 .catalog {
-  display: flex;
+  display: grid;
+  grid-template: 300px / repeat(4, 200px);
   flex-wrap: wrap;
```
Comment (OlyaZaiets) [inline src/styles/_catalog.scss:4]: We don't need flex-wrap, the display is grid

File: src/styles/main.scss [hunk 4]
```diff
@@ -10,5 +10,8 @@ body {
   box-sizing: border-box;
 }
```
Comment (OlyaZaiets) [inline src/styles/main.scss:12]: It's better to move from main.scss

File: src/styles/_catalog.scss [hunk 5]
```diff
@@ -1,22 +1,25 @@
...
-
+  gap: 48px;
+  justify-content: center;
+  place-items: center;
   max-width: 944px;
+  padding-bottom: 48px;
   margin: 0 auto;
```
Comment (OlyaZaiets) [inline src/styles/_catalog.scss:10]: we can center by the grid characteristics

File: src/styles/_catalog.scss [hunk 6]
```diff
@@ -1,22 +1,25 @@
...
 
+  @include media(1024px, 3);
+  @include media(768px, 2);
+  @include media(488px, 1);
```
Comment (OlyaZaiets) [inline src/styles/_catalog.scss:14]: According to the task, four columns starting at 1024px, but you have 3, the same situation with 768, 488.
image

**Comment by OlyaZaiets** [comment]
Good job, need some modification.

File: src/styles/_catalog.scss (hunk 3 above)
Comment (Anna-Tkachenko) [reply src/styles/_catalog.scss:4]: agree, delete redundant property

File: src/styles/_catalog.scss (hunk 6 above)
Comment (Anna-Tkachenko) [reply src/styles/_catalog.scss:14]: fix @OlyaZaiets comment

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
Check @OlyaZaiets comments and fix them

**Review by Anna-Tkachenko — CHANGES_REQUESTED** [review]
There should be 1 column below `488px`, check it
image

File: src/styles/blocks/_catalog.scss [hunk 7]
```diff
@@ -0,0 +1,44 @@
+.catalog {
+  display: grid;
+  grid-template: 300px / repeat(4, 200px);
```
Comment (alexandra-protyanova) [inline src/styles/blocks/_catalog.scss:3]:
```suggestion
  grid-template: 300px / repeat(4, $card-width);
```

File: src/styles/blocks/_catalog.scss [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+  place-items: center;
+  max-width: 944px;
+  padding-bottom: 48px;
+  margin: 0 auto;
+
+  @media (max-width: 488px) {
+    & {
+      grid-template-columns: repeat(1, $card-width);
+    }
+  }
```
Comment (alexandra-protyanova) [inline src/styles/blocks/_catalog.scss:15]: we can make this case with one card default for our page

Comment (alexandra-protyanova) [reply src/styles/blocks/_catalog.scss:15]: And do we need `repeat` for one card?

**Review by alexandra-protyanova — CHANGES_REQUESTED** [review]

File: src/styles/main.scss [hunk 9]
```diff
@@ -1,5 +1,11 @@
 html {
   font-family: Roboto, sans-serif;
+
+  @media (max-width: 488px) {
```
Comment (alexandra-protyanova) [inline src/styles/main.scss:4]: You don't need this media for `html`

File: src/styles/blocks/_catalog.scss [hunk 10]
```diff
@@ -1,19 +1,13 @@
 .catalog {
   display: grid;
-  grid-template: 300px / repeat(4, 200px);
+  grid-template: 300px / repeat(4, $card-width);
```
Comment (alexandra-protyanova) [inline src/styles/blocks/_catalog.scss:3]: Now you have default for 4 card repeat, why do you need media for this?

**Review by alexandra-protyanova — CHANGES_REQUESTED** [review]
Check how your catalog looks on `487px`

**Review by alexandra-protyanova — APPROVED** [review]

**Pull request closed without merging**